src/main/java/org/sagebionetworks/bridge/validators/AssessmentValidator.java

```diff
@@ -0,0 +1,64 @@
+package org.sagebionetworks.bridge.validators;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.sagebionetworks.bridge.BridgeConstants.BRIDGE_EVENT_ID_ERROR;
+import static org.sagebionetworks.bridge.BridgeConstants.BRIDGE_EVENT_ID_PATTERN;
+
+import java.util.Optional;
+
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import org.sagebionetworks.bridge.models.OperatingSystem;
+import org.sagebionetworks.bridge.models.assessments.Assessment;
+
+public class AssessmentValidator implements Validator {
+
+    static final String CANNOT_BE_BLANK = "cannot be missing, null, or blank";
+    private final Optional<Assessment> optional;
+    
+    public AssessmentValidator(Optional<Assessment> optional) {
+        this.optional = optional;
+    }
+    
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return Assessment.class.isAssignableFrom(clazz);
+    }
+
+    @Override
+    public void validate(Object target, Errors errors) {
+        Assessment assessment = (Assessment)target;
+        
+        if (isBlank(assessment.getGuid())) {
+            errors.rejectValue("guid", CANNOT_BE_BLANK);   
+        }
+        if (isBlank(assessment.getAppId())) {
+            errors.rejectValue("appId", CANNOT_BE_BLANK);   
+        }
+        if (isBlank(assessment.getTitle())) {
+            errors.rejectValue("title", CANNOT_BE_BLANK);   
+        }
+        if (isBlank(assessment.getOsName())) {
+            errors.rejectValue("osName", CANNOT_BE_BLANK);   
+        } else if (!OperatingSystem.ALL_OS_SYSTEMS.contains(assessment.getOsName())) {
+            errors.rejectValue("osName", "is not a supported platform");
+        }
+        if (isBlank(assessment.getIdentifier())) {
+            errors.rejectValue("identifier", CANNOT_BE_BLANK);
+        } else if (optional.isPresent()) {
+            errors.rejectValue("identifier", "already exists in revision " + assessment.getRevision());
```

Comment:
Instead of having this obscure indirect check for whether the assessment revision is already used, it would probably be cleaner to have AssessmentService throw when it finds that the assessment has already been used.

Reply:
See comment above. In the course of refactoring I decided this could be removed. Throwing EntityAlreadyExists instead.